Fix: TensorFlow eager example with TensorFlow 2.X

nuka137 opened the pull request:
This PR fixes #565 .

**Summary**

* Eager execution mode is default in TF 2.X. So, removed `tf.enable_eager_execution()` which is not supported in TF 2.X.
* The optimization APIs are changed in TF 2.X. New APIs are as follows.
  * `tf.optimizers.RMSprop`
  * `tf.optimizers.Adam`
  * `tf.optimizers.SGD`

toshihikoyanase commented:
In addition, we can raise errors like [this example](https://github.com/optuna/optuna/blob/master/examples/chainer_simple.py#L31-L32) if users use tensorflow<2.0.0. 
I think such a change is optional, and we can work on it in a follow-up PR.

nuka137 commented:
@toshihikoyanase 

Thanks for your review!
I fixed the issues you mentioned.
Also, I added version checker at the top of `examples/tensorflow_eager_simple.py`.
Could you review again?

P.S.
I also noticed that there is a example which uses TensorFlow Estimator (`tensorflow_estimator_simple.py`).
I think APIs provided by TensorFlow Estimator is also changed in TF 2.0.
In my environment, `tensorflow_estimator_simple.py` also raises error because of API change.
I will tackle this problem as well ASAP.
Thanks.

toshihikoyanase commented on examples/tensorflow_eager_simple.py:
```diff
@@ -20,15 +20,18 @@
 
 import tensorflow as tf
 from tensorflow.keras.datasets import mnist
+import pkg_resources
```
Please reorder the imports in alphabetical order.

https://circleci.com/gh/optuna/optuna/26154
```
flake8 .
./examples/tensorflow_eager_simple.py:23:1: H306: imports not in alphabetical order (tensorflow.keras.datasets.mnist, pkg_resources)
1     H306: imports not in alphabetical order (tensorflow.keras.datasets.mnist, pkg_resources)
```

nuka137 replied:
Thanks! I fixed it.

c-bata commented on examples/tensorflow_eager_simple.py:
```diff
@@ -51,7 +54,7 @@ def create_model(trial):
-    optimizer_options = ['RMSPropOptimizer', 'AdamOptimizer', 'MomentumOptimizer']
+    optimizer_options = ['RMSprop', 'Adam', 'SGD']
     optimizer_selected = trial.suggest_categorical('optimizer', optimizer_options)
     if optimizer_selected == 'RMSPropOptimizer':
```
Hi @nuka137 ! I'm not familiar with tensorflow, but it looks these conditions to set optimizer's arguments should be updated like:

```python
if optimizer_selected == 'RMSProp':
    kwargs['learning_rate'] = trial.suggest_loguniform('rmsprop_learning_rate', 1e-5, 1e-1)
    ...
elif optimizer_selected == 'Adam':
    ...
elif optimizer_selected == 'SGD':
    ...
```

toshihikoyanase replied:
@c-bata 
Good catch! Thank you.

nuka137 replied:
@c-bata 

Thanks! I forgot to fix these.

@toshihikoyanase 

I also fixed the parameter names for `suggest_loguniform`.
Could you check them, please?

nuka137 commented:
Also I fixed TensorFlow estimator's example.
The patch is now in [my repository](https://github.com/nuka137/optuna/commit/9be0c90f7acf886bdd5fc961d0fb4c4fe2325779).
Should this merge into this PR or create new PR?

nuka137 commented:
@toshihikoyanase 

Yes, I agree with you.
I'll create new PR for it after I fix your issue.

toshihikoyanase commented:
@nuka137 Thank you for creating #868.
I'll merge this PR after #868 to avoid CI failure.

hvy commented:
#868 is not merged but I noticed that we must also update https://github.com/optuna/optuna/blob/master/examples/pruning/tensorflow_estimator_integration.py. Let me take a look at it.
